features/282-next-repo-re-organization/plan.md

```diff
@@ -0,0 +1,63 @@
+# Re-organize Github Repo to focus on NEXT
+
+## Repository layout
+
+The top level is the git repository.
+
+- skygear-server
+    - gateway
+    - auth gear
+    - CF gateway and related
+- skycli
+- skygear-SDK-JS
+- skygear-SDK-iOS
+- skygear-SDK-Android
+- skygear-controller
```

Comment:
second thought, we can't use controller without platform anyway right? So why not just merge it to `skygear-server` too?

Reply:
The controller will only work with the db and k8s, it will not interact with the platform directly and it is quite standalone. But you are right, controller needs platform. It needs the gateway db schema and tenant config structure. We can merge it to `skygear`, but see if we want to make it public. It was private component in v1.

Reply:
Hmmm you are right, it is a bit tricky as for portal backend, I think it should be private.

But for skycli backend (and cf controller), they need to be open source for the community edition to actually function right?

Can we move portal related code to another repo, and keep controller a open source repo or inside the main repo?

Reply:
I thought the controller should be the endpoint of both portal and skycli (at least for EE and cloud version), most of their function overlap (signup, login, custom domain...etc.). cli and portal are just the ui.

For CE, the controller maybe another program that we can open source? The features are quite different, no user signup, no custom domain management...etc. Maybe we will not use k8s too, not decided yet. I think we can design this later when we do CE version.